zkevm-circuits/src/rlp_circuit.rs

```diff
@@ -0,0 +1,2368 @@
+                    cb.require_equal(
+                        "tag_index::next == value - 0xf7",
+                        meta.query_advice(rlp_table.tag_index, Rotation::next()),
+                        meta.query_advice(byte_value, Rotation::cur()) - 247.expr(),
+                    );
+                    cb.require_zero(
+                        "length_acc == 0",
+                        meta.query_advice(length_acc, Rotation::cur()),
+                    );
+                },
+            );
+
+            // if tag_index == tag_length && tag_length == 1
+            cb.condition(
+                is_prefix(meta) * tindex_eq_tlength.clone() * tlength_eq.clone(),
+                |cb| {
+                    cb.require_equal("191 < value", value_gt_191.is_lt(meta, None), 1.expr());
+                    cb.require_equal("value < 248", value_lt_248.is_lt(meta, None), 1.expr());
+                    cb.require_equal(
+                        "length_acc == value - 0xc0 (1)",
+                        meta.query_advice(length_acc, Rotation::cur()),
+                        meta.query_advice(byte_value, Rotation::cur()) - 192.expr(),
+                    );
+                },
+            );
+
+            // if tag_index < tag_length && tag_length > 1
+            cb.condition(
+                is_prefix(meta) * tindex_lt_tlength.clone() * tlength_lt.clone(),
+                |cb| {
+                    cb.require_equal(
+                        "length_acc == (length_acc::prev * 256) + value",
+                        meta.query_advice(length_acc, Rotation::cur()),
+                        meta.query_advice(length_acc, Rotation::prev()) * 256.expr()
+                            + meta.query_advice(byte_value, Rotation::cur()),
+                    );
+                },
+            );
+
+            //////////////////////////////////////////////////////////////////////////////////////
+            ////////////////////////////////// RlpTxTag::Nonce ///////////////////////////////////
+            //////////////////////////////////////////////////////////////////////////////////////
+            cb.condition(is_nonce(meta), |cb| {
+                // tag_index < 10
+                cb.require_equal(
+                    "tag_index < 10",
+                    tag_index_lt_10.is_lt(meta, None),
+                    1.expr(),
+                );
+
+                // tag_index >= 1
+                cb.require_zero(
+                    "1 <= tag_index",
+                    not::expr(or::expr([tindex_lt.clone(), tindex_eq.clone()])),
+                );
+            });
+
+            // if tag_index == tag_length && tag_length == 1
+            cb.condition(
+                is_nonce(meta) * tindex_eq_tlength.clone() * tlength_eq.clone(),
+                |cb| {
+                    cb.require_equal("value < 129", value_lt_129.is_lt(meta, None), 1.expr());
```

Comment:
It should be `value < 0x80 = 128`. Because if `value == 0x80`, then `value_acc = 0` instead of 0x80. And this case should never happen for "Nonce".

Reply:
Note that "Nonce = 0" can be rlp-encoded into `0x80`. You can an example [tx](https://etherscan.io/getRawTx?tx=0xbce17823a5f6400cd3e0222cfa9c57974d9b158d990fcf1471c4d47ad9f661b5) here.